apps/condo/domains/organization/components/DeleteEmployeeButtonWithReassignmentModel.tsx

```diff
@@ -0,0 +1,245 @@
+                onFinally: () => {
+                    setIsDeleting(false)
+                },
+                intl,
+            },
+        )
+    }
+    
+    const handleDeleteAndReassignTicketsClick = async () => {
+        setIsDeleting(true)
+        setIsConfirmVisible(false)
+
+        let updatedTicketsCount = 0
+
+        while (updatedTicketsCount < activeTicketsOrganizationEmployeeCount) {
+            notificationApi.warning({
+                message: notificationTitle(NotificationTitleWarningLabel),
+                description: notificationMessage(NotificationMessageWarningLabel, updatedTicketsCount),
+                duration: 0,
+                key: 'reassignTicket',
+            })
+
+            try {
+                const { data: ticketsToReassign } = await loadTicketsToReassign()
+
+                if (isEmpty(ticketsToReassign?.tickets)) break
+                const { data: reassignedTickets }  = await updateTickets({
+                    variables: {
+                        data: ticketsToReassign?.tickets?.map(ticket => ({
+                            id: ticket?.id,
+                            data: getTicketReassignData(ticket),
+                        })),
+                    },
+                })
+
+                updatedTicketsCount += reassignedTickets?.tickets?.length
+                await sleep(1000)
+            } catch (err) {
+                if (errorLoadTickets || errorUpdateTickets || err) {
+                    notificationApi.error({
+                        message: notificationTitle(NotificationTitleErrorLabel),
+                        description: notificationMessage(NotificationMessageErrorLabel),
+                        duration: 0,
+                        key: 'reassignTicket',
+                    })
+                }
+                setIsDeleting(false)
+                return
+            }
+        }
+
+        runMutation(
+            {
+                action: softDeleteAction,
+                onError: (e) => { throw e },
+                OnErrorMsg: () => ({
+                    message: notificationTitle(NotificationTitleErrorLabel),
+                    description: notificationMessage(NotificationMessageErrorLabel),
+                    duration: 0,
+                    key: 'reassignTicket',
+                }),
+                onFinally: () => setIsDeleting(false),
+                OnCompletedMsg: () => ({
+                    message: notificationTitle(NotificationTitleSuccessLabel),
+                    description: notificationMessage(NotificationMessageSuccessLabel, updatedTicketsCount),
+                    duration: 0,
+                    key: 'reassignTicket',
+                }),
+                intl,
+            },
+        )
+    }
+
+    // TODO: DOMA-10834 add search for an employee along with specialization
+    const search = useMemo(() => {
+        return searchEmployeeUser(employeeOrganizationId, (organizationEmployee: OrganizationEmployee) => {
```

Comment:
There are wrong types here, but it's ok for now. Will need to be fixed when these search utilities are updated.

Not in this pr

Reply:
Okey, thanks for comment